### PR TITLE
Fix the notations for dynamic_update_slice and dynamic_slice

### PR DESCRIPTION
**Pull request by sdasgup3**
`adjusted_start_indices` seems like a 1-d tensor from the type of `shape(operand)` in `clamp(0, start_indices, shape(operand) -slice_sizes)`.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -2237,7 +2237,7 @@ contain the sizes of the slice for each dimension.
 
 More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` where:
 
-* `jd = adjusted_start_indices[d][] + id`.
+* `jd = adjusted_start_indices[d] + id`.
```
Comment (burmako) [inline docs/spec.md:2240]: Given that `start_indices` is a vararg of 0-dimensional tensors, I imagine that `adjusted_start_indices` has the same type? If that's the case, then we need the second `[]` to turn a 0-dimensional tensor into a scalar, right?

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec.md (hunk 1 above)
Comment (sdasgup3) [reply docs/spec.md:2240]: @burmako 
Thanks! I agree with why we should need.

Having noted that and assuming that `clamp` is same as `stablehlo.clamp`, there is still some confusion on the usage of `clamp` op  as  `stablehlo.clamp` which takes  arguments of the same shape, where as we have a var arg of `tensor<T>` (for `start_indices`) and `tensor<1xT>` (for `shape`). 


If you agree to that then we can add it to be done as part of #289.

Comment (ghpvnist) [inline docs/spec.md:2240]: `id = adjusted_start_indices[d] + jd` like below?

**Review by ghpvnist — CHANGES_REQUESTED** [review]

File: docs/spec.md (hunk 1 above)
Comment (sdasgup3) [reply docs/spec.md:2240]: I also doubted the same before, but fortunately this formulation is correct. So we should not need this.

Comment (sdasgup3) [reply docs/spec.md:2240]: The way I look at it is: 
index of larger window (the operand in this case) = start indices + the indices of the slice (the result here)

Comment (sdasgup3) [reply docs/spec.md:2240]: Please let me know if this makes sense.

**Review by burmako — APPROVED** [review]

File: docs/spec.md (hunk 1 above)
Comment (burmako) [reply docs/spec.md]: Sounds like a plan!

**Review by ghpvnist — APPROVED** [review]

**Comment by sdasgup3** [comment]
@burmako Can you help merge it?

**Pull request merged**